### PR TITLE
Added callback to handle printer custom components in printer backend.

### PR DESCRIPTION
**Pull request by akallabeth**
* Move handling of device redirection channel custom components to backend.
* Save settings of each local printer, restore them (no distinction between remote hosts, only printers)

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

File: channels/printer/client/printer_main.c [hunk 1]
```diff
@@ -71,6 +73,417 @@ struct _PRINTER_DEVICE
...
+	if (printer_config_valid(oldPath))
+		rc = MoveFileA(oldPath, newPath);
+
+	free(oldPath);
+	free(newPath);
+	return rc;
+}
+
+#if 0
+sprintf_s(printer_dev->port, sizeof(printer_dev->port), "PRN%d", printer->id);
```
Comment (hardening) [inline channels/printer/client/printer_main.c]: I guess this section should be removed ?

File: channels/printer/client/printer_main.c [hunk 2]
```diff
@@ -293,6 +706,182 @@ static UINT printer_irp_request(DEVICE* device, IRP* irp)
...
+						Stream_Seek(s, NewPrinterNameLen);
+
+						if (!printer_move_config(settings, OldPrinterName, OldPrinterNameLen, NewPrinterName,
+						                         NewPrinterNameLen))
+							return ERROR_INTERNAL_ERROR;
+					}
+					break;
+
+				default:
+					WLog_ERR(TAG, "Unknown chache data eventID: 0x%08"PRIX32"", eventID);
```
Comment (hardening) [inline channels/printer/client/printer_main.c]: Typo: chache => cache

**Pull request merged**